### PR TITLE
Accept and send DTMF via data channel

### PR DESCRIPTION
**Pull request by dennwc**
Accept and send DTMF via data channel. Also take the chance to make data callbacks and functions accept an interface for data messages.

Blocked on https://github.com/livekit/protocol/pull/630

### CODE REVIEW

File: localparticipant.go [hunk 1]
```diff
@@ -269,42 +269,107 @@ func (p *LocalParticipant) closeTracks() {
...
+}
+
+type DataPacket interface {
+	ToProto() *livekit.DataPacket
+}
+
+var (
+	_ DataPacket = UserData(nil)
+	_ DataPacket = (*UserDataPacket)(nil)
+	_ DataPacket = (*livekit.SipDTMF)(nil) // implemented in the protocol package
```
Comment (biglittlebigben) [inline localparticipant.go:312]: Could you comment on why these definitions are needed?

Comment (dennwc) [reply localparticipant.go:312]: Done

**Comment by dennwc** [comment]
Hmm, right, the integration test needs an updated `livekit-server` image. So it will have to wait for https://github.com/livekit/livekit/pull/2559.

**Comment by dennwc** [comment]
Removed test for SIP messages for. It will be tested in SIP repo anyway. Ready for final review now.

**Review by biglittlebigben — APPROVED** [review]

**Pull request merged**